src/spacel/agent/instance.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import time
+import urllib2
+
+logger = logging.getLogger('spacel')
+
+
+class InstanceManager(object):
+    """
+    Perform actions directly on this instance.
+    """
+
+    def __init__(self, back_off_scale=0.1):
+        self._back_off_scale = back_off_scale
+
+    def health(self, manifest, timeout=120):
+        """
+        Healthcheck an endpoint from the manifest for timeout in s
+        :param manifest:  Manifest.
+        :param timeout:  Timeout.
+        :return: Health status (boolean)
+        """
+        if not manifest.healthcheck.get('endpoint'):
+            return True
+
+        retries = 0
+        start = time.time()
+        while True:
+            if self.check_health(manifest.healthcheck.get('endpoint'),
```

Comment:
Maybe `._check_health()`? Feels like private helper to `.health()`.

Maybe pass `manifest.healthcheck` as an argument, then unpack inside?
Prevents duplicated defaults (i.e. `localhost` on :31 and :46), and could de-dent for style.